compiler-core/src/analyse.rs

```diff
@@ -109,24 +109,47 @@ pub fn infer_module(
         register_type_alias(t, &mut type_names, &mut env, &name)?;
     }
 
+    // Check name uniqueness among constants, functions and external functions.
+    let mut name_locations = HashMap::new();
+    for c in &statements.constants {
+        name_locations
+            .entry(&c.name)
+            .or_insert_with(Vec::new)
+            .push(c.location)
+    }
+    for f in &statements.functions {
+        name_locations
+            .entry(&f.name)
+            .or_insert_with(Vec::new)
+            .push(f.location)
+    }
+    for ef in &statements.external_functions {
+        name_locations
+            .entry(&ef.name)
+            .or_insert_with(Vec::new)
+            .push(ef.location)
+    }
+    for (name, locations) in name_locations {
+        let mut sorted_locations = locations.clone();
+        sorted_locations.sort_by_key(|l| l.start);
+        for location in sorted_locations {
+            assert_unique_name(&mut value_names, name, location)?;
+        }
+    }
+
```

Comment:
Rather than having this extra looping code we can order the two spans in `error.rs` just before printing them. Something like this:

```rust
let (first, second) = if a.start < b.start {
    (a, b)
} else {
    (b, a)
};
```